tcms/xmlrpc/serializer.py

```diff
@@ -121,15 +121,60 @@ def serialize_queryset(self):
         Returns: List
         """
         response = []
-        for m in self.queryset:
-            self.model = m
-            m = self.serialize_model()
-            response.append(m)
+        for model in self.queryset:
+            self.model = model
+            model = self.serialize_model()
+            response.append(model)
 
         del self.queryset
         return response
 
 
+def _get_single_field_related_object_pks(m2m_field_query, model_pk, field_name):
+    field_names = []
+    for item in m2m_field_query[model_pk]:
+        if item[field_name]:
+            field_names.append(item[field_name])
+    return field_names
+
+
+def _get_related_object_pks(m2m_fields_query, model_pk, field_name):
+    """Return related object pks from query result via ManyToManyFields
+
+    Any object pk with value 0 or None values will be excluded in the final
+    list.
+
+    :param dict m2m_field_query: the result returned from _query_m2m_fields
+    :param model_pk: whose object's related object pks will be retrieved
+    :type model_pk: int or long
+    :param str field_name: field name of the related object
+    :return: list of related objects' pks
+    :rtype: list
+    """
+    data = m2m_fields_query[field_name]
+    return _get_single_field_related_object_pks(data, model_pk, field_name)
+
+
+def _serialize_names(row, values_fields_mapping):
+    # Replace name from ORM side to the serialization side as expected
```

Comment:
Turn this comment into a docstring for this function. Doesn't need to describe all the parameters like docstrings for RPC methods but at least shows what is happening.